mathesar_ui/src/stores/table-data/processedColumns.ts

```diff
@@ -103,7 +103,7 @@ export function processColumn({
       linkFk ? linkFk.referent_table : undefined,
       abstractType.cellInfo,
     ),
-    allowedFiltersMap: getFiltersForAbstractType(abstractType.identifier),
+    allowedFiltersMap: getFiltersForAbstractType(abstractType.identifier, linkFk),
```

Comment:
The newly created utility function should be called here, instead of calling `getFiltersForAbstractType` directly.